### PR TITLE
Plugin to decompress Zip and RAR files

### PR DESCRIPTION
**Pull request by ianstalk**
This is a plugin for extracting Zip/RAR files within Flexget with an interface similar to the "move" plugin. To extract RAR files, it requires the rarfile Python package, and RARLabs Unrar, both of which are available for free.

### CODE REVIEW

File: flexget/plugins/plugin_rarfile.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
+from __future__ import unicode_literals, division, absolute_import
+import logging
+import os
+import rarfile
```
Comment (cvium) [inline flexget/plugins/plugin_rarfile.py]: I think it'd be best to have some conditional importing so as not to break everything if `rarfile` is not installed.

Comment (ianstalk) [inline flexget/plugins/plugin_rarfile.py]: Good point. I've fixed that in my latest commit.

File: flexget/plugins/plugin_rarfile.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+from flexget import plugin
+from flexget.entry import Entry
+from flexget.event import event
+from flexget.utils.template import render_from_entry, RenderError
+
+try:
+    import rarfile
+except ImportError:
+    pass
```
Comment (gazpachoking) [inline flexget/plugins/plugin_rarfile.py]: You can just set `rarfile = None` here, then test for that at runtime instead of the current hack. :wink:

**Comment by gazpachoking** [comment]
Seems good to me. Would it be better though to make it a general extraction plugin? Perhaps this could be used https://pypi.python.org/pypi/pyunpack instead of rarfile. (not vouching for the quality of that particular library, just an idea)

**Comment by gazpachoking** [comment]
It would also be cool to leverage #473 (once we figure out exactly how that should work) in here to rename the contents once they are extracted.

**Comment by ianstalk** [comment]
I'll look into it. At the very least, it should be simple to add ZIP support since the rarfile API intentionally mirrors the built-in zipfile module. That'd get us 80% there in terms of what archives you're likely to encounter.

**Comment by ianstalk** [comment]
Pyunpack and the underlying patool package it uses seem pretty limited it what they can do. You can extract a complete archive, but there's no way to extract files individually. regexp filtering and smart renaming would be out.

**Comment by gazpachoking** [comment]
Cool, zip and rar sound good enough to me then.

**Comment by gazpachoking** [comment]
Sweet! Want to make a wiki page?

**Comment by gazpachoking** [comment]
Also, does this work in a task that has download plugin to extract the downloaded thing? If not it would be nice if we could make it.

**Comment by gazpachoking** [comment]
Also, we should make sure the priorities are okay so that it can work combined with move plugin.

**Pull request merged**